packages/google_sign_in/lib/testing.dart

```diff
@@ -0,0 +1,79 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'package:flutter/services.dart' show MethodCall;
+
+/// A fake backend that can be used to test components that require a valid
+/// [GoogleSignInAccount].
+///
+/// Example usage:
+///
+/// GoogleSignIn googleSignIn;
+/// FakeSignInBackend fakeSignInBackend;
+///
+/// setUp(() {
+///   googleSignIn = new GoogleSignIn();
+///   fakeSignInBackend = new FakeSignInBackend();
+///   fakeSignInBackend.user = new FakeUser(
+///     id: 123,
+///     email: 'jdoe@example.org',
+///   );
+///   googleSignIn.channel.setMockMethodCallHandler(
+///       fakeSignInBackend.handleMethodCall);
+/// });
+///
+class FakeSignInBackend {
+  FakeUser user;
+
+  /// Handles method calls that would normally be sent to the native backend.
+  /// Returns with the expected values based on the current [user].
+  Future<dynamic> handleMethodCall(MethodCall methodCall) async {
+    switch (methodCall.method) {
+      case 'init':
+        // do nothing
+        return null;
+      case 'getTokens':
+        return <String, String>{
+          'idToken': user.idToken,
+          'accessToken': user.accessToken
+        };
+      case 'signIn':
+        return user._asMap;
+      case 'signInSilently':
+        return user._asMap;
+      case 'disconnect':
+        user = null;
```

Comment:
Hmm, you probably want `user` to remain set since it represents the user in the backing store (not whether the user is signed in).

Given that the actual plugin maintains signed-in state, I think you can avoid maintaining any notion of sign-in state here and just return the empty map without modifying `user`